spec/models/meter_spec.rb

```diff
@@ -1,6 +1,33 @@
 require 'rails_helper'
 
+class DummyMeterAttributes
+
+  ATTRIBUTES = {
+      meter_corrections: [{ readings_start_date: Date.new(2010, 6, 25), }],
+      storage_heaters: [{ start_date: Date.new(2010, 1, 1) }],
+      solar_pv: [ 'yay']
+    }
+
+  def self.for(mpan_mprn = nil, area_name = nil, meter_type = nil)
+    ATTRIBUTES
+  end
+end
+
 describe 'Meter', :meters do
+  describe 'meter attributes' do
+    let(:meter) { build(:electricity_meter) }
+
+    it 'can get them all' do
+      meter =  build(:electricity_meter)
+      expect(meter.meter_attributes(DummyMeterAttributes)).to eq DummyMeterAttributes::ATTRIBUTES
+    end
+
+    it 'drops storage heaters if env set' do
```

Comment:
Change method call to receive ENV as a parameter with a default of env and use env.fetch